### ISSUE TITLE
Non-Cautionary Accidental not rendered in presence of 2nd Staff

### ISSUE THREAD

**Issue by arshiacont** [grame-cncm/guidolib#81]
Note the following snippet with the rendered result:
```
{
[ \staff<2>  
   (* meas. 1 *)  \clef<"f4"> \key<2> \meter<"C", autoBarlines="off", autoMeasuresNum="system"> \stemsUp \beamsOff {d#-1/4, d#0/4  } \stemsUp \beamsOff {e-1/4, e0/4  } \stemsUp \beamsOff {c#-1/4, c#0/4  } \stemsUp \beamsOff {d-1/4, d0/4  } ]
  }
```
![image](https://user-images.githubusercontent.com/3298204/60198145-3d48fb00-9841-11e9-83e0-7360b0140b3f.png)

Now, add a second staff and note how the _Natural_ is not rendered on the last chord of the 2nd staff:
```
{[ \staff<1>   \barFormat<style= "system", range="1"> 
   (* meas. 1 *)  \clef<"g2"> \key<2> \meter<"C", autoBarlines="off", autoMeasuresNum="system"> \stemsDown \slurBegin:1<curve="up"> \beamBegin:1 \accent<position="above">( b2/16)
 \stemsDown a/16 \slurEnd:1 \stemsDown \slurBegin:1<curve="up"> a/16 \stemsDown f#/16 \slurEnd:1 \beamEnd:1 \stemsDown \slurBegin:1<curve="up"> \beamBegin:1 a/8 \stemsDown g/8 \slurEnd:1 \beamEnd:1 \stemsDown \slurBegin:1<curve="up"> \beamBegin:1 a/16 \stemsDown g/16 \slurEnd:1 \stemsDown \slurBegin:1<curve="up"> g/16 \stemsDown e/16 \slurEnd:1 \beamEnd:1 \stemsDown \slurBegin:1<curve="up"> \beamBegin:1 g/8 \stemsDown f#/8 \slurEnd:1 \beamEnd:1 ]
 , 
[ \staff<2>  \barFormat<style= "system", range="2"> 
   (* meas. 1 *)  \clef<"f4"> \key<2> \meter<"C", autoBarlines="off", autoMeasuresNum="system"> \stemsUp \beamsOff {d#-1/4, d#0/4  } \stemsUp \beamsOff {e-1/4, e0/4  } \stemsUp \beamsOff {c#-1/4, c#0/4  } \stemsUp \beamsOff {d-1/4, d0/4  } ]
  }
```
![image](https://user-images.githubusercontent.com/3298204/60198640-6027df00-9842-11e9-9053-d2d16c5ff2e7.png)

**Comment by arshiacont** [comment]
After investigation, if you remove `\barFormat<style= "system", range="1"> ` from staff 1 the problem disappears.

This is just a reduced simple from a multi-instrument score. So I need to keep separate ranges... .

The conflict appears to come from the `style="system"`.

In case it helps.

**Comment by dfober** [comment]
Fixed !

**Comment by arshiacont** [comment]
Thanks @dfober . Almost there but not yet... :) 

Try this (same excerpt as above but in context):
```
{[ \staff<1> \barFormat<style= "system", range="1"> 
   \clef<"g2"> \key<2> \meter<"C"> \stemsDown \slurBegin:1<curve="up"> \beamBegin:1 \accent<position="above">( b2/16)
 \stemsDown a/16 \slurEnd:1 \stemsDown \slurBegin:1<curve="up"> a/16 \stemsDown f#/16 \slurEnd:1 \beamEnd:1 \stemsDown \slurBegin:1<curve="up"> \beamBegin:1 a/8 \stemsDown g/8 \slurEnd:1 \beamEnd:1 \stemsDown \slurBegin:1<curve="up"> \beamBegin:1 a/16 \stemsDown g/16 \slurEnd:1 \stemsDown \slurBegin:1<curve="up"> g/16 \stemsDown e/16 \slurEnd:1 \beamEnd:1 \stemsDown \slurBegin:1<curve="up"> \beamBegin:1 g/8 \stemsDown f#/8 \slurEnd:1 \beamEnd:1 \bar 
   ]
, 
[ \staff<2> \accol<id=0, range="2-3"> \barFormat<style= "system", range="2-3"> 
   \clef<"g2"> \key<2> \meter<"C"> _/8 \stemsUp \beamBegin:1 {b0/8, f#1/8, a/8  } _/8 \stemsUp {b0/8, e1/8, g/8  } \beamEnd:1 _/8 \stemsUp \beamBegin:1 {a0/8, e1/8, g/8  } _/8 \stemsUp {a0/8, d1/8, f#/8  } \beamEnd:1 \bar 
] , 
[ \staff<3> 
   \clef<"f4"> \key<2> \meter<"C"> \stemsUp \beamsOff {d#-1/4, d#0/4  } \stemsUp \beamsOff {e-1/4, e0/4  } \stemsUp \beamsOff {c#-1/4, c#0/4  } \stemsUp \beamsOff {d-1/4, d0/4  } \bar 
]  }
```

Note that the last chord on 3rd staff should be Natural.

Leads to:
![image](https://user-images.githubusercontent.com/3298204/61324117-f469de00-a811-11e9-92e0-2d91567e258a.png)

instead of 
<img width="277" alt="image" src="https://user-images.githubusercontent.com/3298204/61324142-00ee3680-a812-11e9-9863-c37622198776.png">

**Comment by arshiacont** [comment]
*Hint:* If I move the `\bar` on each measure/staff, the accidental is actually rendered!

The problem is ofcourse when the `\bar` is there such as in here:

```
{[ \staff<1> \barFormat<style= "system", range="1"> 
   (* meas. 1 *)  \clef<"g2"> \key<2> \meter<"C", autoBarlines="off", autoMeasuresNum="system"> \stemsDown \slurBegin:1<curve="up"> \beamBegin:1 \accent<position="above">( b2/16)
 \stemsDown a/16 \slurEnd:1 \stemsDown \slurBegin:1<curve="up"> a/16 \stemsDown f#/16 \slurEnd:1 \beamEnd:1 \stemsDown \slurBegin:1<curve="up"> \beamBegin:1 a/8 \stemsDown g/8 \slurEnd:1 \beamEnd:1 \stemsDown \slurBegin:1<curve="up"> \beamBegin:1 a/16 \stemsDown g/16 \slurEnd:1 \stemsDown \slurBegin:1<curve="up"> g/16 \stemsDown e/16 \slurEnd:1 \beamEnd:1 \stemsDown \slurBegin:1<curve="up"> \beamBegin:1 g/8 \stemsDown f#/8 \slurEnd:1 \beamEnd:1 \bar 
   (* meas. 2 *)  \stemsDown \beamBegin:1 \accent<position="above">( g/16)
 \stemsDown f#/16 \stemsDown e/16 \stemsDown d/16 \beamEnd:1 \stemsUp \beamBegin:1 c#/16 \stemsUp b1/16 \stemsUp a/16 \stemsUp g/16 \beamEnd:1 \stemsUp \slurBegin:1<curve="down"> \beamBegin:1 f#/8 \stemsUp d2/8 \slurEnd:1 \stemsUp \slurBegin:1<curve="down"> d/8 \stemsUp c#/8 \slurEnd:1 \beamEnd:1 ]
 , 
[ \staff<2> \accol<id=0, range="2-3"> \barFormat<style= "system", range="2-3"> 
   (* meas. 1 *)  \clef<"g2"> \key<2> \meter<"C", autoBarlines="off", autoMeasuresNum="system"> _/8 \stemsUp \beamBegin:1 {b0/8, f#1/8, a/8  } _/8 \stemsUp {b0/8, e1/8, g/8  } \beamEnd:1 _/8 \stemsUp \beamBegin:1 {a0/8, e1/8, g/8  } _/8 \stemsUp {a0/8, d1/8, f#/8  } \beamEnd:1 \bar 
   (* meas. 2 *)  _/8 \stemsUp \beamBegin:1 {g0/8, c#1/8, g/8  } _/8 \stemsUp {e/8, g/8, c#2/8  } \beamEnd:1 _/8 \stemsUp \beamBegin:1 {d1/8, f#/8, b/8  } _/8 \stemsUp {e/8, f#/8, a#/8  } \beamEnd:1 ]
 , 
[ \staff<3> \staffFormat<dy=-7.6> \barFormat<style= "system", range="3"> 
   (* meas. 1 *)  \clef<"f4"> \key<2> \meter<"C", autoBarlines="off", autoMeasuresNum="system"> \stemsUp \beamsOff {d#-1/4, d#0/4  } \stemsUp \beamsOff {e-1/4, e0/4  } \stemsUp \beamsOff {c#-1/4, c#0/4  } \stemsUp \beamsOff {d-1/4, d0/4  } \bar 
   (* meas. 2 *)  \stemsUp \beamsOff {e-1/4, e0/4  } \stemsUp \beamsOff {e-1/4, e0/4  } \stemsUp \beamsOff {f#-1/4, f#0/4  } \stemsUp \beamsOff {f#-1/4, f#0/4  } ]
  }
```

**Comment by dfober** [comment]
fixed (with issue #144)
